src/DataLayouts/broadcast.jl

```diff
@@ -1,3 +1,9 @@
+import MultiBroadcastFusion as MBF
+
+# Make a MultiBroadcastFusion type, `FusedMultiBroadcast`, and macro, `@fused`:
+# via https://github.com/CliMA/MultiBroadcastFusion.jl
+MBF.@make_fused FusedMultiBroadcast fused
```

Comment:
Would you mind leaving the original definition of this struct here, without the macro annotation? I think it's fine to define iteration utilities and frontend macros in `MultiBroadcastFusion.jl`, but this sort of core struct definition should probably be explicitly written out in ClimaCore, as it was in previous commits to this PR. Also, we could potentially use other frontends to this struct; i.e., with more fusion capability than MBF.

Reply:
Sure, did you have something specific in mind? Is it alright if I leave it as a comment for now? I just figured that it would be nice to not require users to define macros and deal with hygiene. Also, since it defines the struct, we could make different types and `@fused` macros.

The "core" function of MBF is `fused_pairs`, but I think we could add more options (https://github.com/CliMA/MultiBroadcastFusion.jl/issues/5).

Reply:
cc @dennisYatunin?

Reply:
Sure, you can leave this a comment for now to get some performance measurements. We can remove the `make_fused` definition and replace it with one that supports fusion across center+face spaces in a future PR.

Reply:
So, `make_fused` is technically compatible with fusion across center+face spaces, those checks/that functionality have to happen on the ClimaCore side. `make_fused` is currently limited in that it can't contain loops, (generally) function calls, aliases, and if-else statements. I agree that we can try this in a future PR.